crates/store/README.md

```diff
@@ -0,0 +1,829 @@
+     over two type parameters, `K` and `V`, which represent the key and value
+     types respectively.
+
+  2. **`DBIter`**: This trait defines the core functionality that any
+     database-specific iterator must implement. It includes methods for moving
+     to the next item (`next()`) and reading the current value (`read()`). In
+     this way it is similar to Rust's `Iterator` trait.
+
+  3. **`Structured` and `Unstructured`**: These types are used to differentiate
+     between iterators that work with typed, structured data and those that work
+     with raw byte slices.
+
+### Iterator functionality
+
+The `Iter` struct provides several important methods:
+
+  1. **`keys()`**: Returns an iterator over just the keys.
+
+  2. **`entries()`**: Returns an iterator over key-value pairs.
+
+  3. **`structured_key()`**: Converts an unstructured key iterator into a
+     structured one.
+
+  4. **`structured_value()`**: Converts an unstructured value iterator into a
+     structured one.
+
+These methods allow for flexible processing of data, enabling operations on keys
+alone, on full entries, or on structured representations of the data.
+
+#### Type-safe iteration
+
+One of the most powerful features of the iterator system is its use of Rust's
+type system to provide type-safe iteration. The `Structured` type parameter
+allows the iterator to work with strongly-typed keys and values, while still
+allowing for efficient, low-level storage.
+
+For example, an iterator might be defined as:
+
+```rust
+Iter<Structured<ContextState>, Structured<StateData>>
+```
+
+This type ensures that the iterator will yield keys of type `ContextState` and
+values of type `StateData`, providing compile-time guarantees about the types of
+data being processed.
+
+#### Integration with layers and transactions
+
+The iterator system is deeply integrated with other components of the library:
+
+  1. **Layer integration**: The `ReadLayer` trait includes an iter method that
+     returns an `Iter`. This allows each layer to provide its own implementation
+     of iteration, potentially adding additional functionality or optimisations.
+
+  2. **Transaction integration**: The `Transaction` struct provides methods for
+     creating iterators over its operations. This allows for efficient
+     processing of pending changes in a transaction.
+
+#### Range queries
+
+The iterator system supports efficient range queries through the use of a
+`start` key in the `iter()` method of `ReadLayer`. This allows for iteration to
+begin at a specific point in the key space, enabling efficient partial scans of
+the data.
+
+#### Performance considerations
+
+The iterator system is designed with performance in mind:
+
+  1. **Lazy evaluation**: Iterators are lazy, meaning they only do work when
+     asked for the next item. This allows for efficient processing of large
+     datasets without unnecessary memory usage.
```

Comment:
Also allows you to stop at will, even though the iterator is only created with `start`